### PR TITLE
fix: Defaulting emphasis input/serialization to _ from *

### LINKED ISSUE #59: Rich text editor can generate potentially invalid commonmark
Reported on [Meta.SE](https://meta.stackexchange.com/a/360136/395497)

**Describe the bug**

> Overlapping bold/italic formatting leads to incorrect formatting when toggling to Markdown mode.
> 
> 1. In rich text mode, write some text.
>    * this is bold, now italic, and now?!
> 2. Select part of the text.
> 3. Make the selection italic.
>    * this is bold, *now italic, and now?!*
> 4. Select some *overlapping* text.
> 5. Make the selection bold.
>    * **this is bold, _now italic_**_, and now?!_
> 
> (Now bold and italic formatting is correctly applied to overlapping portions of the text.)
> 
> When switching to Markdown, the formatting is screwed up because Markdown doesn’t support overlapping formatting in this way: adjacent `*`/`**` is ambiguous at the best of times, but Markdown wouldn’t support this even when using `_` to format italics.
> 
> Here’s what the editor generates:
> 
> ```markdown
> **this is bold, *now italic****, and now?!*
> ```
> 
> … but as mentioned, even the following isn’t really supported by Markdown as far as I’m aware (it certainly isn’t on this site):
> 
> ```markdown
> **this is bold, _now italic**, and now?!_
> ```

### PR DESCRIPTION
**Pull request by tmcentee**
Closes #59 . 

**Describe your changes**

This simply changes the default Markdown serialization for emphasis (italics) from `*` to `_`. This now allows text with overlapping marks (e.g. bold and italics) to roundtrip between rich text and markdown successfully. 

emphasis

In rich-text mode, either `_` or `*` can be used to mark text with emphasis. Both will serialize to `_` in Markdown.
auto-format

**PR Checklist**

- [ ] All new/changed functionality includes unit and (optionally) e2e tests as appropriate
- [ ] All new/changed functions have `/** ... */` docs
- [ ] I've added the `bug`/`enhancement` and other labels as appropriate

**Environment(s) tested**

- Device: Desktop
- OS: Win10
- Browser: Chrome
- Version: 103

### CODE REVIEW

**Comment by tmcentee** [comment]
cc @b-kelly I know you had commented on the reporting Meta post about some edge cases you can think of. Let me know if you still have any in mind that I'll need to check out here. 

https://meta.stackexchange.com/questions/360033/opt-in-alpha-test-for-a-new-stacks-editor/360136#comment1202381_360136

File: test/shared/markdown-serializer.test.ts [hunk 1]
```diff
@@ -78,6 +78,10 @@ describe("markdown-serializer", () => {
             `<a href="https://www.example.com">https://www.example.com</a>`,
             `<https://www.example.com>`,
         ],
+        [
+            "<strong>Bold </strong><em><strong>Both </strong></em><em>Italic</em>",
```
Comment (b-kelly) [inline test/shared/markdown-serializer.test.ts:82]: The html generated by the editor is:

```html
<p><strong>This is bold, </strong><em><strong>this is both</strong>, this is emphasis</em></p>
```

Note the em wrapping both. Ideally, we should unit test this in a more controlled manner by using the exact doc that is generated in this state:

```json
{
    "type": "doc",
    "attrs": {
        "markup": ""
    },
    "content": [
        {
            "type": "paragraph",
            "attrs": {
                "markup": ""
            },
            "content": [
                {
                    "type": "text",
                    "marks": [
                        {
                            "type": "strong",
                            "attrs": {
                                "markup": ""
                            }
                        }
                    ],
                    "text": "This is bold, "
                },
                {
                    "type": "text",
                    "marks": [
                        {
                            "type": "em",
                            "attrs": {
                                "markup": ""
                            }
                        },
                        {
                            "type": "strong",
                            "attrs": {
                                "markup": ""
                            }
                        }
                    ],
                    "text": "this is both"
                },
                {
                    "type": "text",
                    "marks": [
                        {
                            "type": "em",
                            "attrs": {
                                "markup": ""
                            }
                        }
                    ],
                    "text": ", this is emphasis"
                }
            ]
        }
    ]
}
```

File: src/shared/markdown-serializer.ts [hunk 2]
```diff
@@ -597,7 +597,12 @@ const extendedCodeMarkDeserializer: typeof defaultCodeMarkDeserializer = {
 // extend the default markdown serializer's marks and add our own
 const customMarkdownSerializerMarks: MarkdownSerializerMarks = {
     ...defaultMarkdownSerializer.marks,
-    em: genMarkupAwareMarkConfig(defaultMarkdownSerializer.marks.em),
+    em: genMarkupAwareMarkConfig({
```
Comment (b-kelly) [inline src/shared/markdown-serializer.ts:600]: If we change the default here, we also need to change it in the commonmark editor. Otherwise, we'll just end up in the same boat.

**Review by b-kelly — CHANGES_REQUESTED** [review]
I'll try and remember what those additional edge cases were. I should have written them down 😅. In the meanwhile, I have a few change requests.

File: src/shared/markdown-serializer.ts (hunk 2 above)
Comment (tmcentee) [reply src/shared/markdown-serializer.ts:600]: I assume that's handled by this change: https://github.com/StackExchange/Stacks-Editor/pull/161/files#diff-98cdb9c85b4c07d8f39aa6d21d0934ca06651821fbbf3a774d55b09ce04df62dR687

**Comment by tmcentee** [comment]
@b-kelly This has been orphaned for a while. Still on the radar to eventually ship or should I close?
